Mathlib/Data/Seq/Seq.lean

```diff
@@ -336,24 +337,32 @@ theorem eq_of_bisim (bisim : IsBisimulation R) {s₁ s₂} (r : s₁ ~ s₂) : s
         And.imp id (fun r => ⟨tail s, tail s', by cases s using Subtype.recOn; rfl,
           by cases s' using Subtype.recOn; rfl, r⟩) this
       have := bisim r; revert r this
-      cases' s with x s <;> cases' s' with x' s'
-      · intro r _
-        constructor
-        · rfl
-        · assumption
-      · intro _ this
-        rw [destruct_nil, destruct_cons] at this
-        exact False.elim this
-      · intro _ this
-        rw [destruct_nil, destruct_cons] at this
-        exact False.elim this
-      · intro _ this
-        rw [destruct_cons, destruct_cons] at this
-        rw [head_cons, head_cons, tail_cons, tail_cons]
-        obtain ⟨h1, h2⟩ := this
-        constructor
-        · rw [h1]
-        · exact h2
+      cases s with
+      | nil =>
+        cases s' with
+        | nil =>
+          intro r _
+          constructor
+          · rfl
+          · assumption
+        | cons x' s' =>
+          intro _ this
+          rw [destruct_nil, destruct_cons] at this
+          exact False.elim this
+      | cons x s =>
+        cases s' with
+        | nil =>
+          intro _ this
+          rw [destruct_nil, destruct_cons] at this
+          exact False.elim this
+        | cons x' s' =>
+          intro _ this
+          rw [destruct_cons, destruct_cons] at this
+          rw [head_cons, head_cons, tail_cons, tail_cons]
+          obtain ⟨h1, h2⟩ := this
+          constructor
+          · rw [h1]
+          · exact h2
```

Comment:
I don't really like how this is becoming significantly longer.

Reply:
@urkud [said on Zulip](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/The.20plan.20to.20remove.20cases'/near/500222566) that it's OK if there are no shorter options.

Reply:
https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/cases'.20in.20Data.2FSeq.2FSeq

Reply:
@j-loreaux I've committed what should be a better replacement, based on [this](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/cases'.20in.20Data.2FSeq.2FSeq/near/502693277).